tapestry.js

```diff
@@ -862,6 +907,111 @@ function setupMedia(id, mediaFormat, mediaType, mediaUrl, width, height) {
     return mediaEl;
 }
 
+// Builds the view mode, including functionality to
+function changeToViewMode(adjustedOn) {
+    // For calculating where the node should go
+    var nodeRatio = getBrowserWidth() / getTapestryDimensions().width;
+    var nodeRadius = NORMAL_RADIUS * nodeRatio;
+    var nodeSpace = (nodeRadius * 2) + ((nodeRadius * 2) * 0.10);
+
+    var children = getChildren(root);
+    var coordinates = [];
+    for (var i = 0; i < children.length; i++) {
+        if (adjustedOn === "width") {
+            if (i % 2 === 0) {
+                coordinates[children[i]] = {
+                    "fx": 0,
+                    "fy": Math.min(getTapestryDimensions().height * (i / (children.length - 1)) + nodeRadius, getTapestryDimensions().height - nodeRadius)
```

Comment:
Since you are using `children.length - 1` here, if the length is 1, this would be 0 which would mean you are dividing by 0.

Reply:
Oh yeah, I think that can be changed to just children.length

Reply:
Okay yeah that fixed the single node issue! Yay

Reply:
Awesome!